README.md

```diff
@@ -2,16 +2,18 @@
 
 [![CI](https://github.com/readchina/WikidataLookup/actions/workflows/ci.yml/badge.svg)](https://github.com/readchina/WikidataLookup/actions/workflows/ci.yml)
 
-This repo contains a Python package for verifying named entities in [ReadAct](https://github.com/readchina/ReadAct).
+This repo contains a Python package for verifying the authenticity of named entities basing on [ReadAct](https://github.com/readchina/ReadAct).
 
-The goal is to automately extract data about **person**, **space**, and **institutions** from ReadAct's data tables, and to use [Wikidata](https://www.wikidata.org/wiki/Wikidata:Main_Page) for verification and to extract missing data points.
+The goal is to automately extract data about **person**, **space**, and **institutions** from ReadAct database, and to use [Wikidata](https://www.wikidata.org/wiki/Wikidata:Main_Page) to verify existing data and update missing data.
 
-## Requierments
+The two way of using it are either install it locally and run the fowllowing scripts, or commit-push your updates for ReadAct to ReadAct where this package will be ran in CI automatically.
 
-- Python: `>=3.8`
+## Requierments
 
 The tool is tested on macOS and linux. 
 
+Python: `>=3.8`
+
 Also you need to upgrade pandas by using the command:
 
 `python -m pip install --upgrade pip pandas`
```

Comment:
if you rebase your PR I m not sure if we still need the `--upgrade` command to pandas, I'd really like to understand better where these requirements originate

Reply:
I have searched and also tested in different ways. Here are my findings:
1.    The ReadActor CI has the following step to instal dependencies (includes `pandas`)
```
      - name: Install App dependencies
        uses: py-actions/py-dependency-install@v3
        with:
          path: "requirements.txt"
```
Similarly, in ReadMe, user is told to do `pip install -r requirements.txt` manually after install the package which is uploaded to PyPi.
Which is to say, no matter locally when we use ReadActor or it is run on its own CI, the dependencies' installation are separated from the python project installation. 

2.    In `ci.yml` of ReadAct, currently we have:
```
- name: Use pip Cache  
        uses: actions/cache@v2
        with:
          path: ~/.cache/pip
          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
          restore-keys: |
            ${{ runner.os }}-pip-
```
I am not 100% confident with this code section, but I have checked we don't have any `requirements.txt` in ReadAct. Does it mean that it can read the requirements.txt in installed packages like ReadActor and frictionless? But it is run before install ReadActor and frictionless, I don't think this block is working.

3.    What we have tried ```python -m pip install --upgrade pip pandas``` means that we want pandas to be installed. I have tested with or without the flag '--upgrade` which means to install the latest version are both fine.

4.    I searched the differences between define the requirements in `requirements.txt` and `setup.py`, the result is that: if we are going to update the tool ReadActor and are going to distribute in on PyPi, we need both `requirements.txt` and `setup.py` for ReadActor, where `requirements.txt` is expected to have specific version requirements but `setup.py` can either have more abstract require like `install_requires=["Click", "pandas", "requests"]` or more detailed `install_requires=["Click", "pandas>=1.5.0", "requests>=2.26"]`.

> [Whereas install_requires metadata is automatically analyzed by pip during an install, requirements files are not, and only are used when a user specifically installs them using python -m pip install -r.](https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/?spm=a2c6h.12873639.article-detail.9.1c5b71d3vsdlzc#id4)

I have forked ReadActor, made it a new project, edited the setup.py, uploaded it to Test PyPi, with the goal that pandas will be installed automatically as a dependency. The project can be uploaded successfully but the install is not. Since there is no `pandas 1.5.0` on Test PyPi, I am now guessing it will only work with the real PyPi. 

```
pip3.8  install -i https://test.pypi.org/simple/ ReadActorDep==1.0.2 
Looking in indexes: https://test.pypi.org/simple/
Collecting ReadActorDep==1.0.2
  Downloading https://test-files.pythonhosted.org/packages/50/ba/7acb681fa18e12ba2cd7dd5e9a7986feb2ae7a5347e6a772ce86b00ee330/ReadActorDep-1.0.2-py3-none-any.whl (843 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 843.7/843.7 kB 638.4 kB/s eta 0:00:00
Collecting Click
  Using cached https://test-files.pythonhosted.org/packages/c2/f1/df59e28c642d583f7dacffb1e0965d0e00b218e0186d7858ac5233dce840/click-8.1.3-py3-none-any.whl (96 kB)
ERROR: Could not find a version that satisfies the requirement pandas>=1.5.0 (from readactordep) (from versions: none)
ERROR: No matching distribution found for pandas>=1.5.0
```